bashtop

```diff
@@ -425,6 +443,40 @@ quit_() { #? Clean exit
 	exit 0
 }
 
+update_ ()  {  #  update label  menu  to   upgrade  to new version   
+     
+    local -a  root_binary_path=("/bin/" "/usr/bin/")  
+    local  src=$(basename $0)
+    local  path 
+    
+    # the update   menu  is also  availabe  inside cloned  repository 
+    # if bashtop  is installed  in  the system  the update   menu will be also available 
+    #+ to get the lastest feature or  new update   version   
+    if [[  ! -d .git ]]  ; then
+        for  xbin in ${root_binary_path[@]} ; do
+              [[  -x ${xbin}${src}  ]]&& path=${xbin}${src} 
+        done 
+    else 
+        path=${src} 
+    fi
+    
+    [[ -n  ${path}  ]] && {
+         curl  ${remote_src}  -o ${path} 
+         test $? -eq  0 && {
+                [[ -x  ${root_binary_path[1]}notify-send ]]  && {
+                     notify-send  --urgency=normal   "${src}  updated to v${version}"   
```

Comment:
Whats the availability of `notify-send` over different distributions and future cross platform compatibility?

Reply:
Most GNU/Linux distributions have this command by default 
but the notification appears only if it is available.  
and don't break the script